internal/query/relations.go

```diff
@@ -717,9 +943,24 @@ func (e *engine) GetRole(ctx context.Context, roleResource types.Resource) (type
 			actions[i] = relationToAction(action)
 		}
 
+		dbRole, err := e.db.GetRoleByID(ctx, roleResource.ID)
+		if err != nil && !errors.Is(err, database.ErrNoRoleFound) {
+			e.logger.Error("error while getting role", zap.Error(err))
+		}
+
+		if dbRole == nil {
+			dbRole = new(database.Role)
+		}
```

Comment:
As before, if there's no corresponding CRDB entry for the role, that should be an error. We don't want a mismatch between the state of CRDB and the state of SpiceDB.